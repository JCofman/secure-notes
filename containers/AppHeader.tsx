import React from 'react';

import { PrimaryButton } from 'react-components/components/button';
import Header from 'react-components/components/header/Header';

interface AppHeaderProps {
  isEditing: boolean;
  onCreateNewNote: () => void;
}

const AppHeader = (props: AppHeaderProps) => {
  const { isEditing, onCreateNewNote } = props;

  return (
    <Header className='toolbar flex flex-items-center'>
      <PrimaryButton className='ml1' icon='add' disabled={isEditing} onClick={onCreateNewNote}>
        {' '}
        New note
      </PrimaryButton>
    </Header>
  );
};

export default AppHeader;
