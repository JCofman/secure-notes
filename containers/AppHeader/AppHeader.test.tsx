import React from 'react';
import { render } from '@testing-library/react';

import AppHeader from './AppHeader';

test(`AppHeader is rendered properly`, () => {
    const { container } = render(
        <AppHeader
            isEditing={false}
            onCreateNewNote={() => {
                // do nothing.
            }}
        />
    );
    expect(container).toMatchSnapshot();
});
