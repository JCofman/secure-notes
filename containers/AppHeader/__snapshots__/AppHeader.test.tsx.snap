// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader is rendered properly 1`] = `
<div>
  <header
    class="toolbar flex flex-items-center"
  >
    <button
      aria-busy="false"
      class="pm-button pm-button--primary ml1"
      role="button"
      type="button"
    >
      <svg
        class="icon-16p"
        focusable="false"
        role="img"
        viewBox="0 0 16 16"
      >
        <use
          xlink:href="#shape-note-add"
        />
      </svg>
       
      New note
    </button>
  </header>
</div>
`;
